lib/chef/knife/azurerm_base.rb

```diff
@@ -264,6 +264,18 @@ def msg_server_summary(server)
       end
 
       def validate_params!
+        if locate_config_value(:connection_user).nil?
+          raise ArgumentError, "Please provide --connection-user option for authentication."
+        end
+
+        unless locate_config_value(:connection_password).nil? ^ locate_config_value(:ssh_public_key).nil?
+          raise ArgumentError, "Please specify either --connection-password or --ssh-public-key option for authentication."
+        end
+
+        if is_image_windows? && !locate_config_value(:ssh_public_key).nil?
+          raise ArgumentError, "Windows doesn't support SSH authentication. Please provide --connection-password option."
```

Comment:
Did we tested this? Do you mean windows doesn't support ssh key based authentication ?